src/runtime.js

```diff
@@ -78,82 +78,173 @@ define([
         var nodeType = ast.type;
         var result = defaultResult;
 
-        // TODO order alphabetically
         switch (nodeType) {
-        case "Program":
-            result = this.visitProgram(ast, state);
+        case "ArrayExpression":
+            result = this.visitArrayExpression(ast, state);
             break;
-        case "ExpressionStatement":
-            result = this.accept(ast.expression, state);
+        case "ArrowExpression":
+            console.log("ArrowExpression visitor not implemented yet");
```

Comment:
I would suggest to create a function to log this as a warning, it could look like this:
```
function warnMissingImplementation(elem) {
    console.warn(elem + " not implemented yet");
}
```
And it could be used like this
case "ArrayExpression":
case "ArrowExpression":
// more here
warnMissingImplementation(nodeType);
break;
```